operator/api/redpanda/v1alpha2/role_types.go

```diff
@@ -109,6 +122,14 @@ type RoleStatus struct {
 	// ManagedRole returns whether the role has been created in Redpanda and needs
 	// to be cleaned up.
 	ManagedRole bool `json:"managedRole,omitempty"`
+	// Principals is the last successfully synced list of principals for this role.
+	// It mirrors the normalized spec.principals currently supported; RoleBinding
+	// aggregation is not yet implemented.
+	// Pattern validation ensures entries are in normalized "Type:Name" format.
```

Comment:
All comments, aside from the markers, are turned into user facing documentation. Notes like this don't belong there.

It might be possible to include it after the marker?

```
// +kubebuilder:validation:items:Pattern=`^(User:.+|[^:]+)$` // Ensures usage of normalized format.
```